source/UIManager.cpp

```diff
@@ -0,0 +1,60 @@
+#include "UIManager.hpp"
+
+std::map<Ogre::String, UIOverlay *> UIManager::overlays;
+
+UIManager::UIManager(void)
+{}
+
+void UIManager::init() {
+
+	UIOverlayHUD *hud = new UIOverlayHUD();
```

Comment:
Unmanaged pointers are evil.
Please use ```std::unique_ptr``` or some equivalent.
This probably also applies for he contents of the map where the pointer is inserted.